components/calendar/src/ethiopic.rs

```diff
@@ -232,21 +252,31 @@ impl Ethiopic {
 impl Date<Ethiopic> {
     /// Construct new Ethiopic Date.
     ///
+    /// For the Amete Mihret era style, negative years work with
+    /// year 0 as 1 pre-Incarnation, year -1 as 2 pre-Incarnation,
+    /// and so on.
+    ///
     /// ```rust
     /// use icu::calendar::Date;
+    /// use icu::calendar::ethiopic::EthiopicEraStyle;
     ///
     /// let date_ethiopic =
-    ///     Date::new_ethiopic_date(2014, 8, 25).expect("Failed to initialize Ethopic Date instance.");
+    ///     Date::new_ethiopic_date(EthiopicEraStyle::AmeteMihret, 2014, 8, 25)
+    ///     .expect("Failed to initialize Ethopic Date instance.");
     ///
     /// assert_eq!(date_ethiopic.year().number, 2014);
     /// assert_eq!(date_ethiopic.month().ordinal, 8);
     /// assert_eq!(date_ethiopic.day_of_month().0, 25);
     /// ```
     pub fn new_ethiopic_date(
-        year: i32,
+        era_style: EthiopicEraStyle,
+        mut year: i32,
         month: u8,
         day: u8,
     ) -> Result<Date<Ethiopic>, DateTimeError> {
+        if era_style == EthiopicEraStyle::AmeteAlem {
+            year -= 5493;
```

Comment:
Issue: I think this is the wrong constant.

Nit: Put this constant in a common place rather than burying it down here.